source/views/components/filter/filter-toolbar.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+import * as React from 'react'
+import type {FilterType, ListType, ToggleType, ListItemSpecType} from './types'
+import {StyleSheet, ScrollView, View} from 'react-native'
+import {Toolbar} from '../toolbar'
+import {FilterToolbarButton} from './filter-toolbar-button'
+import {ActiveFilterButton} from './active-filter-button'
+import flatten from 'lodash/flatten'
+
+type Props = {
+	filters: Array<FilterType>,
+	onPopoverDismiss: (filter: FilterType) => any,
+}
+
+export function FilterToolbar({filters, onPopoverDismiss}: Props) {
+	function updateFilter(filter: FilterType, option?: ListItemSpecType) {
+		if (filter.type === 'toggle') {
+			updateToggleFilter(filter)
+		} else if (filter.type === 'list' && option) {
+			updateListFilter(filter, option)
+		}
+	}
+
+	function updateToggleFilter(filter: ToggleType) {
+		let newFilter = filter
+		newFilter.enabled = false
+		onPopoverDismiss(newFilter)
+	}
+
+	function updateListFilter(filter: ListType, option: ListItemSpecType) {
+		let newFilter = filter
+		newFilter.spec.selected = filter.spec.selected.filter(
+			item => item.title !== option.title,
+		)
+		if (newFilter.spec.selected.length === 0) {
+			if (filter.spec.mode === 'OR') {
+				newFilter.spec.selected = newFilter.spec.options
+			}
+			newFilter.enabled = false
+		}
+		onPopoverDismiss(newFilter)
+	}
+
+	const filterToggles = filters.map((filter, index) => (
+		<FilterToolbarButton
+			key={filter.spec.title}
+			filter={filter}
+			isActive={filter.enabled}
+			onPopoverDismiss={onPopoverDismiss}
+			style={index === 0 ? styles.first : null}
+			title={filter.spec.title}
+		/>
+	))
+
+	const allButtons = filters.filter(f => f.enabled).map((filter, i) => {
+		if (filter.type === 'toggle') {
+			return (
+				<ActiveFilterButton
+					key={filter.spec.title}
+					filter={filter}
+					label={filter.spec.label}
+					onRemove={filter => updateFilter(filter)}
+					style={i === 0 ? styles.first : null}
+				/>
+			)
+		} else if (filter.type === 'list') {
+			return filter.spec.selected.map((selected, j) => (
+				<ActiveFilterButton
+					key={selected.title}
+					filter={filter}
+					label={selected.label || selected.title.toString()}
+					onRemove={filter => updateFilter(filter, selected)}
+					style={i === 0 && j === 0 ? styles.first : null}
+				/>
+			))
+		}
+		return null
+	})
+	const activeFilterButtons = flatten(allButtons)
+	const anyFiltersEnabled = filters.some(f => f.enabled)
+
+	return (
+		<View>
+			<ScrollView horizontal={true} showsHorizontalScrollIndicator={false}>
+				<Toolbar>{filterToggles}</Toolbar>
+			</ScrollView>
+			{anyFiltersEnabled && (
+				<ScrollView horizontal={true} showsHorizontalScrollIndicator={false}>
```

Comment:
Can we put `paddingLeft` as `contentContainerStyle` on this ScrollView, instead of doing styles.first? If contentContainerStyle doesn't work, try `style`, please, but I think we want contentContainerStyle.

Reply:
Update: Try the same thing on the filterToggles scrollview, for the same reason?

Reply:
Adding paddingLeft to the ScrollViews makes the buttons cut off when you scroll to the right, which is not what we want. This is why I opted for setting the marginLeft on the first button. Lmk if you want a screenshot

Reply:
Did you try both contentContainerStyle and style? Using padding on of those should scroll the padding off to the side… the other one won't scroll the padding.

Reply:
I've tested contentContainerStyle with paddingLeft, and it seems to do what we want.

<details><summary>screenshot</summary>

![image](https://user-images.githubusercontent.com/464441/43957577-e890bb72-9c6d-11e8-99ac-1b29932f01de.png)
</details>

<details><summary>patch</summary>

```patch
diff --git a/source/views/components/filter/filter-toolbar.js b/source/views/components/filter/filter-toolbar.js
index 83580e77..5ba239c5 100644
--- a/source/views/components/filter/filter-toolbar.js
+++ b/source/views/components/filter/filter-toolbar.js
@@ -41,18 +41,17 @@ export function FilterToolbar({filters, onPopoverDismiss}: Props) {
 		onPopoverDismiss(newFilter)
 	}
 
-	const filterToggles = filters.map((filter, index) => (
+	const filterToggles = filters.map(filter => (
 		<FilterToolbarButton
 			key={filter.spec.title}
 			filter={filter}
 			isActive={filter.enabled}
 			onPopoverDismiss={onPopoverDismiss}
-			style={index === 0 ? styles.first : null}
 			title={filter.spec.title}
 		/>
 	))
 
-	const allButtons = filters.filter(f => f.enabled).map((filter, i) => {
+	const allButtons = filters.filter(f => f.enabled).map(filter => {
 		if (filter.type === 'toggle') {
 			return (
 				<ActiveFilterButton
@@ -60,17 +59,15 @@ export function FilterToolbar({filters, onPopoverDismiss}: Props) {
 					filter={filter}
 					label={filter.spec.label}
 					onRemove={filter => updateFilter(filter)}
-					style={i === 0 ? styles.first : null}
 				/>
 			)
 		} else if (filter.type === 'list') {
-			return filter.spec.selected.map((selected, j) => (
+			return filter.spec.selected.map(selected => (
 				<ActiveFilterButton
 					key={selected.title}
 					filter={filter}
 					label={selected.label || selected.title.toString()}
 					onRemove={filter => updateFilter(filter, selected)}
-					style={i === 0 && j === 0 ? styles.first : null}
 				/>
 			))
 		}
@@ -81,11 +78,19 @@ export function FilterToolbar({filters, onPopoverDismiss}: Props) {
 
 	return (
 		<React.Fragment>
-			<ScrollView horizontal={true} showsHorizontalScrollIndicator={false}>
+			<ScrollView
+				contentContainerStyle={styles.scroller}
+				horizontal={true}
+				showsHorizontalScrollIndicator={false}
+			>
 				<Toolbar>{filterToggles}</Toolbar>
 			</ScrollView>
 			{anyFiltersEnabled && (
-				<ScrollView horizontal={true} showsHorizontalScrollIndicator={false}>
+				<ScrollView
+					contentContainerStyle={styles.scroller}
+					horizontal={true}
+					showsHorizontalScrollIndicator={false}
+				>
 					{activeFilterButtons}
 				</ScrollView>
 			)}
@@ -94,7 +99,7 @@ export function FilterToolbar({filters, onPopoverDismiss}: Props) {
 }
 
 const styles = StyleSheet.create({
-	first: {
-		marginLeft: 10,
+	scroller: {
+		paddingLeft: 10,
 	},
 })
```
</details>

Reply:
or see 6f9ab3356e260293b98d83e951c9460ab5561959, which is the same patch.

Reply:
Oh nice!